sc2ts/info.py

```diff
@@ -193,6 +193,239 @@ def max_descendant_samples(ts, show_progress=True):
+        hide_runlengths=None,
+        hide_extra_rows=None,
+        hide_labels=None,
+        child_label="C",
+        colours=None,
+        edges=None,  # optional, otherwise will be recalculated
+        exclude_stylesheet=None,
+    ):
+        """
+        Create a styled HTML table indicating bases that differ between the parents of
+        a recombination node. This is suitable for display e.g. in a Jupyter notebook
+        using the ``IPython.display.HTML`` function. The table is accessible from the
+        ``.html`` attribute
+
+        :param ts TreeSequence:
+            The tree sequence to which the nodes refer 
+        :param node int:
+            The node ID of the child node, usually a recombination node.
+            This will be placed on the second row of the copying pattern, so that
+            in the most used case of a recombination node with two parents, the
+            child is the middle row.
+        :param show_bases bool:
+            If True, show the allelic state (i.e. ``A``, ``C``,
+            ``G``, ``T``, or `-`) for each position at which the parents differ.
+            If True, do not plot a character, but simply show coloured table cells.
+            If None (default), show an em-dash for deletions, but nothing else.
+        :param hide_extra_rows bool:
+            If True, hide the rows showing site positions, reference alleles, and
+            de-novo mutation state changes. If False or None (default), show these rows.
+        :param hide_runlengths bool:
+            If True, omit the lower bar that indicates adjacent bases as runs of red
+            (or orange for near-adjacent) and tickmarks. If False or None, show this bar.
+        :param hide_labels bool:
+            If False or None (default), label the rows with P0, P1, etc. If True, hide
+            these row labels.
+        :param child_label str:
+            The label to use for the child node. If None (default), use "C".
+        :param colours list:
+            A list of at least 2 hex colours to use. ``colours[0]`` is used for a base in
+            the child that is not present in any parent, ``colours[1]`` for a base that
+            matches the first parent, ``colours[2]`` for the second parent, etc.
+            Default: None, treated as ``["#FC0", "#8D8", "#6AD", "#B9D", "#A88"]``.
+        :param edges EdgeTable:
+            The edges above the child node, sorted by left coordinate. If not given, this
+            will be recalculated from the data.
+        :param exclude_stylesheet bool:
+            If True, exclude the default stylesheet from the HTML output. This is useful
+            simply to save space if you want to include the copying table in a larger HTML
+            document (e.g. a Jupyter notebook) that already has one copying table shown with
+            the standard stylesheet. If False or None (default), include the default stylesheet.
+        """
+        def node_mutations(ts, node):
```

Comment:
Why are these functions embedded in the constructor for the class? Just make them indepdendent top level functions if they don't use state, so they can be unit tested (in principle). Nested functions only really make sense when they are using some of the variables from the scope

Reply:
It wasn't a separate class before: now that it is, I'll put them as static function of the class instead, actually.